src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = [];
+  let initialState = { ...state };
+
+  actions.forEach((action) => {
+    const { type, extraData, keysToRemove } = action;
+
+    if (type === 'addProperties') {
+      initialState = { ...initialState, ...extraData };
+    } else if (type === 'removeProperties') {
+      initialState = { ...initialState };
+
+      keysToRemove.forEach((key) => {
+        delete initialState[key];
+      });
+    } else if (type === 'clear') {
+      initialState = {};
+    } else {
+      throw new Error(`Unknown action type: ${type}`);
```

Comment:
Good job including a default case for error handling in the `switch` statement. This ensures that the function will throw an error if an unknown action type is encountered.